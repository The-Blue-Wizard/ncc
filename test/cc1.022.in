/* struct return */

#include "support.h"

struct s1       /* s1 copied inline */
{
    int a;
    int b;
    int c;
    int d;
};

struct s2       /* s2 copied via __blkcpy */
{
    long a;
    long b;
    long c;
    long d;
    long e;
};

struct s1 s1_func(struct s1 in)
{
    in.a += 1;
    in.b += 2;
    in.c += 3;
    in.d += 4;

    return in;
}

struct s2 s2_func(struct s2 in)
{
    in.a += 2;
    in.b += 3;
    in.c += 4;
    in.d += 5;
    in.e += 6;

    return in;
}

int
main(void)
{
    struct s1 s1 = { 1, 2, 3, 4 }, s1_ret;
    struct s2 s2 = { 1, 2, 3, 4, 5 }, s2_ret;

    s1_ret = s1_func(s1);
    s2_ret = s2_func(s2);

    putobj(&s1, sizeof(s1));
    putobj(&s1_ret, sizeof(s1_ret));
    putobj(&s2, sizeof(s2));
    putobj(&s2_ret, sizeof(s2_ret));

    return 0;
}
