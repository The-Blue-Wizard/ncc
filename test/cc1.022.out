01000000020000000300000004000000
02000000040000000600000008000000
01000000000000000200000000000000030000000000000004000000000000000500000000000000
03000000000000000500000000000000070000000000000009000000000000000B00000000000000
